code/Controllers/HomeController.cs

```diff
@@ -42,24 +43,41 @@ public HomeController(BagsApi bagsApi)
 			{
 				var product = await bagsApi.GetProduct(productId ?? 0);
 
-				ViewData["url"] = $"https://bagcupid.com/app/{path.TrimStart('/')}";
+				model.url = $"https://bagcupid.com/app/{path.TrimLeadingSlash()}";
 
-				if (product.Images != null && product.Images.Count() > 0)
-					ViewData["image"] = product.Images
+				if (product?.Images != null && product.Images.Count() > 0)
+					model.image = product.Images
 						.Aggregate((selectedImage, nextImage) => (nextImage.Priority < selectedImage.Priority) ? nextImage : selectedImage)
 						.Large;
 			}
 
-			return View("index");
+			return View("index", model);
 		}
 	}
 
+	public class MetaViewModel
```

Comment:
Style: Nesting this inside of `HomeController` will give some context/clarity when instantiating it as to what it is a model for.  At the moment, we only have one controller, but if we add more controllers then having multiple objects named `ViewModel` will quickly lead to confusion as to which one is which.  By nesting inside of `HomeController` instantiating the model will look like `HomeController.ViewModel`, which is more clear as to what the view model is for.

That being said, the name should also include details about the action that the model is for.  In this case, it is for the `App` action so maybe we should name it `AppViewModel`?  Then instantiation would look like `HomeController.AppViewModel`, letting the reader immediately know what exactly this view model is used for.

Side note: What does the `Meta` mean in your name?